### DISCUSSION TITLE
Need help understanding code

### DISCUSSION THREAD

**Discussion by drlauridsen** [tillsteinbach/WeConnect-mqtt#251] [Q&A]
@tillsteinbach 
I am trying to modify your code, to generate autodiscovery sensors and switches for Home Assistant.
I am a beginner level programmer, so I need some help 🙂
What I need to do is get the VIN, and then generate mqtt topics with a specific structure that automatically creates sensors in HA.
I can't seem to find the code where you generate the mqtt topics (or if it is in weconnect mqtt base.py I simply don't understand it), could you point me in the right direction?

**Comment by tillsteinbach** [comment]
WeConnect-mqtt only contains the MQTT part. The rest is in the python library: https://github.com/tillsteinbach/WeConnect-python

**Reply by drlauridsen** [reply]
Since the custom weconnect integration for Home Assistant no longer works, I am again trying to make home assistant autodiscovery for sensors and switches work with WeConnectMqtt.

Just to be sure that I understand the above reply..
The mqtt topics names, including VIN, are not  generated in WeConnect-Mqtt but in WeConnect-Python?
So if I want to create a new group of topics that is called homeassistant/*/* I need to create them in the WeConnect-Python files?

And they are all a part of the docker image?

Sorry if these are very basic questions, Html and php I understand, but python, mqtt and docker are quite new to me

**Reply by tillsteinbach** [reply]
WeConnect-python is the underlying API to talk with the servers. So if you want to extend mqtt it is better to add a module to WeConnect-MQTT that would be loaded when enabled by a command line argument. In that module you could listen for events from WeConnect and then generate your own topics based on this.

**Comment by andlo** [comment]
> add a module to WeConnect-MQTT that would be loaded when enabled by a command line argument.

I were thinking the same, and wanted to do so, but not enough time.
But if it were done, I am sure a lot of VW owners and Homeassistant users would be happy.

One add on to install, and all sensors etc. autodisclvered into Homeassistant :)
